### PR TITLE
Gossip pr 2

### PR DESCRIPTION
**Pull request by art-gor**

### CODE REVIEW

File: src/protocol/gossip/impl/gossip_core.cpp [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+#include <cassert>
+
+#include <libp2p/protocol/gossip/impl/connectivity.hpp>
+#include <libp2p/protocol/gossip/impl/local_subscriptions.hpp>
+#include <libp2p/protocol/gossip/impl/message_builder.hpp>
+#include <libp2p/protocol/gossip/impl/remote_subscriptions.hpp>
+
+namespace libp2p::protocol::gossip {
+
+  // clang-format off
```
Comment (kamilsa) [inline src/protocol/gossip/impl/gossip_core.cpp]: why turn it off here?

Comment (art-gor) [reply src/protocol/gossip/impl/gossip_core.cpp]: to pervent clang-format obfuscation of lambdas

File: include/libp2p/protocol/gossip/gossip.hpp [hunk 2]
```diff
@@ -76,6 +76,9 @@ namespace libp2p::protocol::gossip {
       const ByteArray &from;
       const TopicList &topics;
       const ByteArray &data;
+
+      Message(const ByteArray &f, const TopicList &t, const ByteArray &d)
```
[the remark this answers is not in the document]
Comment (art-gor) [reply include/libp2p/protocol/gossip/gossip.hpp]: will try to get rid of this explicit ctor at all

File: src/protocol/gossip/impl/gossip_core.cpp (hunk 1 above)
Comment (kamilsa) [reply src/protocol/gossip/impl/gossip_core.cpp]: you do that too much in this file

**Pull request merged**